src/metaopt/io/database/mongodb.py

```diff
@@ -19,95 +20,121 @@
     from urllib import quote_plus
 
 
+
 class MongoDB(AbstractDB):
     def __init__(self, dbhost='localhost', dbname='metaopt', dbport=27017, \
                                             username=None, password=None):
         self.dbhost = str(dbhost)
         self.dbname = str(dbname)
         self.username = str(username)
         self.password = str(password)
-
-        # we might have a different port for the database port
-        # so here we do a check if the port is there
-
         self.dbport = str(dbport)
 
         # set the connection string
         # this should be mongodb://mongodb0.example.net:27017
         # according to MongoDB documentation
         # https://docs.mongodb.com/getting-started/python/client/
 
-        self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
-        #self.uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self.username), \
-        #                                      quote_plus(self.password), \
-        #                                     self.dbhost, self.dbport)
         # try to initialize connection
         # setting this as a function allows to make sure reconnection
         # is properly done in the case of an unreliable network
 
         self.conn = self.initiate_connection()
         self.db = self.conn[self.dbname]
 
+        if __debug__:
+            self.logger = logging.getLogger('database')
+            self.fh = logging.FileHandler('database.log')
+            self.fh.setLevel(logging.DEBUG)
+            formatter = logging.Formatter('%(asctime)s - %(name)s -\
+                                        %(levelname)s - %(message)s')
+            self.fh.setFormatter(formatter)
+            self.logger.addHandler(self.fh)
+
+    @property
+    def uri(self):
+        return self._uri
+
+    @uri.setter
+    def uri(self):
```

Comment:
That's not a correct setter interface. It should be 
```python
@uri.setter
def uri(self, value):
```
But that wouldn't make sense since we set it internally anyway. Maybe we just shouldn't have a setter.

Reply:
Gotcha, just a setter would make more sense.

Reply:
I'm kinda confused if we do it this way what `value` should we pass, the value of the URI itself?

Reply:
That's the thing. It would not make sense and that's why I'm saying that maybe there should not be any setter method for uri.

Reply:
So there would only be a `@property`? I'm not that familiar with Python yet, so does it work without a setter? 

In the other case would a `self.uri` as the previous commits be preferable?

Reply:
It works without a setter. Using property can be more flexible if for instance the property is built using other arguments which can change over time. Since uri is built using other arguments I think we should keep @property.